### ISSUE TITLE
java.lang.RuntimeException: Null method binding! void getWorlds(){

### ISSUE THREAD

**Issue by DarkArc** [MinecraftForge/ForgeGradle#368]
``` java
java.lang.RuntimeException: Null method binding! void getWorlds(){
}

        at net.minecraftforge.srg2source.ast.SymbolRangeEmitter.getMethodSignature(SymbolRangeEmitter.java:235)
        at net.minecraftforge.srg2source.ast.SymbolReferenceWalker.visit(SymbolReferenceWalker.java:267)
        at org.eclipse.jdt.core.dom.MethodDeclaration.accept0(MethodDeclaration.java:489)
        at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:2514)
        at net.minecraftforge.srg2source.ast.SymbolReferenceWalker.walk(SymbolReferenceWalker.java:65)
        at net.minecraftforge.srg2source.ast.SymbolReferenceWalker.visit(SymbolReferenceWalker.java:417)
        at org.eclipse.jdt.core.dom.TypeDeclaration.accept0(TypeDeclaration.java:467)
        at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:2514)
        at net.minecraftforge.srg2source.ast.SymbolReferenceWalker.walk(SymbolReferenceWalker.java:65)
        at net.minecraftforge.srg2source.ast.RangeExtractor.generateRangeMap(RangeExtractor.java:162)
        at net.minecraftforge.srg2source.ast.RangeExtractor.generateRangeMap(RangeExtractor.java:88)
        at net.minecraftforge.gradle.tasks.ExtractS2SRangeTask.generateRangeMap(ExtractS2SRangeTask.java:100)
        at net.minecraftforge.gradle.tasks.ExtractS2SRangeTask.doTask(ExtractS2SRangeTask.java:80)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:227)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
        at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
        at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
        at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
        at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
        at org.gradle.internal.Factories$1.create(Factories.java:22)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
        at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
        at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:78)
        at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:48)
        at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:51)
        at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:28)
        at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:43)
        at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:170)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:237)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:210)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
        at org.gradle.launcher.Main.doAction(Main.java:33)
        at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
        at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
```

The build still passes, however, I haven't been able to figure out what's made this appear in my build log.

**Comment by LexManos** [comment]
Something in your code around that function is done oddly so that eclipse can't parse it.
We'd need more full code and context.

**Comment by DarkArc** [comment]
Do you have any suggestion as to how I could find the offending method?

**Comment by LexManos** [comment]
I don't know exactly where off hand but there will be a partial range map file in your gradle temp directory. You can start by reading through that should give you some context.

**Comment by DarkArc** [comment]
So this is what I found:
```
@|com/skelril/skree/content/world/main/MainWorldWrapper.java|218|254|com.skelril.skree.content.world.main|package|com.skelril.skree.content.world.main|(file)
Class Start: com.skelril.skree.content.world.main.MainWorldWrapper
   @|com/skelril/skree/content/world/main/MainWorldWrapper.java|2439|2455|ZoneWaitingLobby|class|com.skelril.skree.content.registry.item.zone.ZoneWaitingLobby
   @|com/skelril/skree/content/world/main/MainWorldWrapper.java|2456|2461|lobby|field|com.skelril.skree.content.world.main.MainWorldWrapper|lobby|
   @|com/skelril/skree/content/world/main/MainWorldWrapper.java|2501|2509|Vector3i|class|com.flowpowered.math.vector.Vector3i
   @|com/skelril/skree/content/world/main/MainWorldWrapper.java|2510|2523|randomizedPos|field|com.skelril.skree.content.world.main.MainWorldWrapper|randomizedPos|
   @|com/skelril/skree/content/world/main/MainWorldWrapper.java|2530|2548|PositionRandomizer|class|com.skelril.nitro.position.PositionRandomizer
   @|com/skelril/skree/content/world/main/MainWorldWrapper.java|2558|2574|createPosition3i|method|com.skelril.nitro.position.PositionRandomizer|createPosition3i|(Lcom/flowpowered/math/vector/Vector3i;)Lcom/flowpowered/math/vector/Vector3i;
   @|com/skelril/skree/content/world/main/MainWorldWrapper.java|2579|2587|Vector3i|class|com.flowpowered.math.vector.Vector3i
   Null method binding! void getWorlds(){
}
```

Which is current to this revision of the file:
https://github.com/Skelril/Skree/blob/ba7829a4e0ca30fbe18ccdff8964ee7a5211fbda/src/main/java/com/skelril/skree/content/world/main/MainWorldWrapper.java

**Comment by LexManos** [comment]
Yes lambdas have issues in the current version of s2s. It's been fixed, and should be rolled out in 1.9.4's version of FG. There are other things I need to change in order for it to work with 1.9.4, so it's a good time to update for J8.

**Comment by DaemonUmbra** [comment]
Closing this issue as it does not appear to be applicable to the currently supported ForgeGradle branch (``FG_3.0``).

If I am incorrect in this regard, please create a new issue prefixed with ``[FG_3.0]`` and containing:
* A summary of the issue (in case the issue has mutated slightly or the original report was poorly described)
* Steps to recreate the issue
* Console output or logs (if possible)

Note that an ongoing effort to update legacy MinecraftForge branches to utilize FG_3.0 is ongoing and accepting informed contributions. (For more information [join the Discord](https://discord.gg/UvedJ9m) and  read through the #mcpconfig channel)
